### ISSUE TITLE
For Typescript

### ISSUE THREAD

**Issue by tmohammad78** [sag1v/react-elastic-carousel#34]
Can I use this package in typescript ? I am refactoring my project to typescript and I should convert it to typescript

**Comment by sag1v** [comment]
Are you having any problems using it in a typescript project?

**Comment by sag1v** [comment]
closing for now. if you got any questions or errors let me know

**Comment by lorenzosjb** [comment]
It says

Could not find a declaration file for module 'react-elastic-carousel'. '/node_modules/react-elastic-carousel/dist/index.js' implicitly has an 'any' type.

Try `npm install @types/react-elastic-carousel` if it exists or add a new declaration (.d.ts) file containing `declare module 'react-elastic-carousel';`ts(7016)

**Comment by sag1v** [comment]
I'm not using TS much so i have little experience with it.
I think we can use (or create) a script that generates the `d.ts` files out of the `propTypes` declarations. I didn't find any tool that doe's that out of the box, I played around with [react-to-typescript-definitions](https://github.com/KnisterPeter/react-to-typescript-definitions) but it seems that we should write a script around that which will actually create the `d.ts` files.

Another option (which i less prefer) is to manually write the `d.ts` files and a script that will generate the `propTypes`.

Anyone wants to take a shot or suggest other alternatives?

**Comment by superjose** [comment]
@sag1v 

Create an `typings.d.ts` file (it can be named anything) in your project's src folder and put:

```ts
declare module 'react-elastic-carousel' {
  export interface ReactElasticCarouselProps {
    className?: string;
    itemsToShow?: number;
// Add more props here!
  }

  declare class Carousel extends React.Component<
    ReactElasticCarouselProps
  > {}
  export default Carousel;
}
```

Then you can do:
`import Carousel from 'react-elastic-carousel'`.

**Comment by sag1v** [comment]
> @sag1v 
> 
> Create an `typings.d.ts` file (it can be named anything) in your project's src folder and put:
> 
> ```ts
> declare module 'react-elastic-carousel' {
>   export interface ReactElasticCarouselProps {
>     className?: string;
>     itemsToShow?: number;
> // Add more props here!
>   }
> 
>   declare class Carousel extends React.Component<
>     ReactElasticCarouselProps
>   > {}
>   export default Carousel;
> }
> ```
> 
> Then you can do:
> `import Carousel from 'react-elastic-carousel'`. 
> 
> 
> 

I'm looking for an automated way to create/update the .d.ts file so it will be synced with the proptypes declarations.

**Comment by sag1v** [comment]
@superjose As discussed, we will start by manually creating the `d.ts` file so we can support typeScript ASAP, and then we will figure out a way to automatically generate and update it based on the lib's proptypes.

Thank you! 🙏

**Comment by sag1v** [comment]
Now available as of version `0.7.0`.
thanks @superjose
